server/main.lua

```diff
@@ -18,17 +16,13 @@ RegisterNetEvent('qb-busjob:server:NpcPay', function()
     local src = source
     local Player = QBCore.Functions.GetPlayer(src)
     if not Player then return end
-    local Payment = math.random(15, 25)
-    if Player.PlayerData.job.name == "bus" then
-        if NearBus(src) then
-            local randomAmount = math.random(1, 5)
-            local r1, r2 = math.random(1, 5), math.random(1, 5)
-            if randomAmount == r1 or randomAmount == r2 then Payment = Payment + math.random(10, 20) end
-            Player.Functions.AddMoney('cash', Payment)
-        else
-            DropPlayer(src, 'Attempting To Exploit')
-        end
-    else
-        DropPlayer(src, 'Attempting To Exploit')
+    if Player.PlayerData.job.name ~= "bus" or not NearBus(src) then return DropPlayer(src, 'Attempting to exploit') end
+
+    local payment = math.random(15, 25)
+    local randomAmount = math.random(1, 5)
+    local r1, r2 = math.random(1, 5), math.random(1, 5)
+    if randomAmount == r1 or randomAmount == r2 then
+        payment += math.random(10, 20)
```

Comment:
I know you are just refactoring this code. But this seems overly complex for generating a random chance of an additional bonus. The probability of a 0.20 chance to get a bonus occurs if you have two tries is 0.36. So we could simplify this to generate a random number between 1 and 100 and check if it's below or equal to 36 to decide to give bonus. All these values should be extracted to config as well. Optional.